### DISCUSSION TITLE
Provide argument `migrate reset` so that it gets used by `prisma db seed`

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [prisma/prisma#21173] [Q&A]
### Question

https://www.prisma.io/docs/guides/migrate/seed-database#user-defined-arguments

I use as input to my seed a json, that is passed in as an argument to the ts file.
How do i pass this to `prisma migrate reset ` so that it gets used by seed ?

```
❯ npx prisma db seed ./STjsons/test_objects_used_by_methods.json
❯ npx prisma migrate reset ./STjsons/test_objects_used_by_methods.json # doesnt work 
```

**Comment by janpio** [comment]
Do I understand correctly that you can provide input to `prisma db seed` and do that in your seed script, but you can not do the same for `migrate reset` (and other commands that use `prisma db seed` under the hood)?

**Reply by LamprosPitsillos** [reply]
Yes that is my issue

**Reply by janpio** [reply]
If I read correctly, `db seed` is technically supposed to support this via `--`:
```
npx prisma db seed -- --environment development
```
https://www.prisma.io/docs/guides/migrate/seed-database#user-defined-arguments:~:text=npx%20prisma%20db%20seed%20%2D%2D%20%2D%2Denvironment%20development

The commands you posted above, do not use that.
How are you accessing these values? Are you using similar code to what is documented for `--`?

(The docs on all this are not super great, sorry.)

**Reply by LamprosPitsillos** [reply]
I have also tried with `--` ,with the same results
Seed working as expected
But reset takes no arguments

**Reply by janpio** [reply]
Can you share the code that you are using to read the value without `--`?

I am aware this does not work, but am collecting the relevant information to formulate and figure out what asolution could look like here. Your unexpected usage (vs. the docs) is relevant input for that.

**Reply by LamprosPitsillos** [reply]
```ts
async function main() {
    const filePath = process.argv[2];
    if (filePath === undefined) {
        console.log("Provide a filePath to the Syntax Tree json file.")
        exit(1)
    }
    const symbol_tree = readClangSyntaxTree(filePath);
    // console.log(symbol_tree.dependencies[0].types);

    // await dropDatabase()
    await fillDatabase(symbol_tree)


}

```

**Reply by janpio** [reply]
Ok, so seems we are not forwarding any argv from `migrate reset` to `db seed`. If we just did that, your solution and the documented one should just start working on `migrate reset` the same way it works on `db seed`.

Can you open a new feature request about this please? Thanks.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
